flytekit/core/promise.py

```diff
@@ -167,6 +181,8 @@ def eval(self) -> bool:
 
         if isinstance(self.rhs, Promise):
             rhs = self.rhs.eval()
+        elif self.rhs.scalar.none_type:
```

Comment:
why is this only added for rhs?  should we also add it for lhs?

Reply:
updated it